### DISCUSSION TITLE
Minimal executable package for a mod

### DISCUSSION THREAD

**Discussion by ygoe** [ramensoftware/windhawk#98] [General]
Is it really necessary to install a 130 MB package that consumes 300 or so MB disk space for a few small mods that consist of not too many lines of source code? A whole IDE and a whole C compiler are distributed here.

I'm thinking towards a leaner distribution way here. Could I, for example, compile a mod to a stand-alone program that I can then deploy to another computer? Maybe also a collection of multiple mods. Maybe even with their current settings fixed if there's no UI for that. And without administrator permissions where they're not needed, if possible. That would make it a lot easier to justify the use of this tool on my work computer and enjoy a consistent behaviour with my personal machines.

7+ Taskbar Tweaker and T-Clock are nice small applications that provide only their features and don't come with all the heavy tools mentioned above.

**Comment by m417z** [comment]
Once you configured Windhawk and its mods, you should be able to remove the Compiler, UI, and AppData\UIData folders which take the vast majority of the space, and just use it. You won't have a way to manage the mods, update them, etc., though. Also, that's not fully supported and tested, so it might not work 100%, but you can try. Let me know how it goes.

**Comment by ygoe** [comment]
I haven't tried it out. But meanwhile I learned that mods have to be recompiled with new symbols that have to be downloaded whenever something changes by a Windows update. So my initial idea probably wouldn't go very far anyway.

**Reply by m417z** [reply]
That's not true, mods don't have to be recompiled.

**Reply by ygoe** [reply]
Hm, what happens when I see the download window that says it's downloading symbols for explorer.exe, and the mod only does its thing again when this download is completed?

**Reply by m417z** [reply]
Symbols are being downloaded when the binaries change, for example after an update. It has nothing to do with code compilation. Why is it an issue?
